### DISCUSSION TITLE
Share your thoughts on Copilot4PRs!

### DISCUSSION THREAD

**Discussion by aspeckofdustin** [community/community#64219] [Pull Requests]
### Copilot in Pull Requests

Product Feedback

### Have thoughts on what parts of the pull request experience could be automated?  We want to hear from you!

As we look to release a more publicly accessibly version of our Copilot4PRs experience, we want to continue to hear feedback from you, our users!  This upcoming experience will have two components in our pull request workflow that we're excited to share with you:

1. **Description summarization** - With Copilot entitlement, you will be able to have Copilot assist you in scanning through your changes to produce a quick summary overview of what your pull request contains to make it easier for a reviewer to catch up on context.
2. **Assisted Code Review** - With Copilot entitlement, you will be able to trigger Copilot to scan through your pull request and return light-weight suggested changes so that your teammate or reviewer can spend more time on deeper suggestions.

We welcome all feedback, both constructive and positive, so please feel free to comment down below with what you would like to see and we will do our best to research into it!

**Comment by Jaber-Saed** [comment]
It's great to hear that you're working on enhancing the pull request experience with Copilot! Automation can play a significant role in streamlining code review and improving collaboration. Here are some thoughts on what parts of the pull request experience could be further automated:

1. **Code Consistency**: Copilot could assist in enforcing coding standards and guidelines by automatically suggesting code changes that align with the project's style. This could reduce the back-and-forth discussions about code formatting.

2. **Security and Best Practices**: Automated checks for security vulnerabilities, code smells, and best practices could help catch potential issues early in the review process.

3. **Documentation Updates**: Copilot could help ensure that any code changes that require updates to documentation (comments, READMEs, API documentation) are automatically identified and suggested.

4. **Changelog Generation**: Automated assistance in generating changelogs based on commit messages and code changes could help maintain clear version histories.

5. **Test Case Suggestions**: Copilot could suggest test cases that cover the changes made in the pull request, promoting better code coverage.

6. **Refactoring Guidance**: If a code change triggers the need for a larger refactoring, Copilot could provide suggestions for how to refactor the affected areas of code.

7. **Suggested Reviewers**: Using historical data and patterns, Copilot could suggest potential reviewers for a pull request, ensuring that the right people are involved.

8. **Integration with CI/CD**: Automated integration with your CI/CD pipeline could provide instant feedback on code changes, including whether they pass tests and adhere to coding standards.

9. **Code Context and Explanation**: Copilot could automatically generate explanatory comments for complex changes, providing context to reviewers.

10. **Repetitive Task Automation**: For tasks like updating dependencies or cleaning up unused code, Copilot could provide automated suggestions or even perform certain actions.

11. **Conflicts Resolution**: Automated assistance in resolving code conflicts during the merging process could save time and prevent manual errors.

12. **Automated Testing**: Copilot could provide suggestions for test cases based on the code changes, helping developers write thorough tests.

Remember, while automation can greatly enhance efficiency, it's important to strike a balance between automation and human decision-making. Providing the option to accept or reject automated suggestions is crucial. Additionally, keeping the communication lines open between developers and reviewers remains important for ensuring high-quality code.

Feedback from users and the developer community will be essential in refining and tailoring these automation features to the specific needs and preferences of different teams and projects.

**Reply by Glen7219** [reply]
Thanks

**Comment by Glen7219** [comment]
How to pull for all public deauth and pentest rquests

**Comment by koen-lee** [comment]
Perhaps Copilot4PRs could also work as a workaround/fix for https://github.com/orgs/community/discussions/9288 .

**Comment by cm121212** [comment]
@aspeckofdustin is there an update on timing for Copilot for PR's early access/beta for existing CfB customers

**Reply by aspeckofdustin** [reply]
@cm121212 Good question!  We're currently working on a landing page to sign up for the Waitlist.  I'll share that here when it's up and running but we're partnering with our Go to Market & Marketing teams to bring that to life currently.  😄

**Comment by cm121212** [comment]
Thanks @aspeckofdustin !  could you share any timing?

**Reply by aspeckofdustin** [reply]
No problem!  It is looking like there will be a landing page in the next week or so.  As soon as it's finalized, I'll circle back and share here.  Thanks for your interest!

**Reply by aspeckofdustin** [reply]
@cm121212 As promised, the last I heard is there will be an official sign up page launching on the first day of our [Universe](https://githubuniverse.com/) event.  That is November 8th, where you'll be able to find the newly launched marketing page and will be able to sign up to be on the waitlist!

**Reply by SnakeO** [reply]
Can you share the sign up page here?

**Comment by 0xdevalias** [comment]
Now that the GitHub Next version of this is getting sunsetted; any chance of having the Copilot for pull requests feature open sourced like TestPilot was?

Guessing probably not, since it would conflict with/undermine some of the 'enterprise only' of the upcoming features.. but can't hurt to ask.

**Reply by cloud26** [reply]
What about third-party pull request copilot? like https://www.codedog.ai/ or https://whatthediff.ai/

**Reply by 0xdevalias** [reply]
To be fair I hadn't looked what alternatives were out there. A quick google led me to these open source repos:

- https://github.com/codedog-ai/codedog
  - > Code review assistant powered by LLM
- https://github.com/Codium-ai/pr-agent
  - > CodiumAI PR-Agent: An AI-Powered Tool for Automated Pull Request Analysis, Feedback, Suggestions and More!
- https://github.com/coderabbitai/ai-pr-reviewer
  - > AI-based Pull Request Summarizer and Reviewer with Chat Capabilities.
  
There may well be other alternatives out there as well.

**Reply by jsunico** [reply]
What happened to Copilot4Prs? Why was it sunsetted?

**Reply by 0xdevalias** [reply]
Because they're rolling it into an enterprise only feature set instead.

**Reply by jsunico** [reply]
Thank you!

**Comment by conorheffron** [comment]
@aspeckofdustin 

It would be nice for `automated release version bumps` (not just packages and libraries), for example updating the release version in a pom.xml for maven or build.gradle for gradle projects with an automated CI/CD build process.

For example, inside the project tags of a pom.xml:
```
<version>2.2.0</version>
```

See sample [here](https://github.com/cph39/ironoc/blob/main/pom.xml), where version tag value is manually tied to release version [here](https://github.com/cph39/ironoc/releases/tag/v2.2.0)
